h/static/scripts/login-forms/components/test/AccountSettingsForms-test.js

```diff
@@ -0,0 +1,293 @@
+  });
+
+  it('renders per-field error messages from js-config', () => {
+    fakeConfig.forms.email.errors = {
+      email: 'email.error.email',
+      password: 'email.error.password',
+    };
+    fakeConfig.forms.password.errors = {
+      password: 'password.error.password',
+      new_password: 'password.error.new_password',
+      new_password_confirm: 'password.error.new_password_confirm',
+    };
+
+    const { elements } = createWrapper();
+
+    assert.equal(
+      elements.emailForm.emailField.prop('fieldError'),
+      fakeConfig.forms.email.errors.email,
+    );
+    assert.equal(
+      elements.emailForm.passwordField.prop('fieldError'),
+      fakeConfig.forms.email.errors.password,
+    );
+    assert.equal(
+      elements.passwordForm.currentPasswordField.prop('fieldError'),
+      fakeConfig.forms.password.errors.password,
+    );
+    assert.equal(
+      elements.passwordForm.newPasswordField.prop('fieldError'),
+      fakeConfig.forms.password.errors.new_password,
+    );
+    assert.equal(
+      elements.passwordForm.confirmNewPasswordField.prop('fieldError'),
+      fakeConfig.forms.password.errors.new_password_confirm,
+    );
+  });
+
+  it('includes the social connect links', () => {
+    const { elements } = createWrapper();
+
+    assert.equal(
+      elements.connectAccountLinks.google.prop('href'),
+      fakeConfig.routes['oidc.connect.google'],
+    );
+    assert.equal(
+      elements.connectAccountLinks.facebook.prop('href'),
+      fakeConfig.routes['oidc.connect.facebook'],
+    );
+    assert.equal(
+      elements.connectAccountLinks.orcid.prop('href'),
+      fakeConfig.routes['oidc.connect.orcid'],
+    );
+  });
+
+  it('indicates when social accounts are already connected', () => {
+    fakeConfig.context.identities.google.connected = true;
+    fakeConfig.context.identities.facebook.connected = true;
+    fakeConfig.context.identities.orcid.connected = true;
+    fakeConfig.context.identities.orcid.url =
+      'https://orcid.org/0000-0002-6373-1308';
+
+    const { elements } = createWrapper();
+
+    assert.equal(elements.connectAccountLinks.google.prop('href'), undefined);
+    assert.equal(elements.connectAccountLinks.facebook.prop('href'), undefined);
+    assert.equal(
+      elements.connectAccountLinks.orcid.prop('href'),
+      'https://orcid.org/0000-0002-6373-1308',
+    );
+  });
+
+  it('omits the social account links when the features are disabled', () => {
+    fakeConfig.features.log_in_with_google = false;
+    fakeConfig.features.log_in_with_facebook = false;
+    fakeConfig.features.log_in_with_orcid = false;
+
+    const { elements } = createWrapper();
+
+    assert.isTrue(elements.connectAccountEl.isEmpty());
+    assert.isTrue(elements.connectAccountLinks.google.isEmpty());
+    assert.isTrue(elements.connectAccountLinks.facebook.isEmpty());
+    assert.isTrue(elements.connectAccountLinks.orcid.isEmpty());
```

Comment:
This triggers a warning about `isEmpty` being a deprecated method. This should be written instead as `assert.isFalse(someElement.exists())`. You can also use `assert.isFalse(wrapper.exists(someSelector))`.